content/blog/migrating-a-cloud-application-to-kubernetes/index.md

```diff
@@ -0,0 +1,391 @@
+                    ],
+                    resources: {
+                        limits: {
+                            memory: "500Mi",
+                            cpu: "500m"
+                        }
+                    }
+                }],
+            }
+        }
+    }}, {
+    provider: eksCluster.provider,
+});
+
+const serversideListener = new k8s.core.v1.Service("server-side-listener", {
+    metadata: { labels: serverDeployment.metadata.labels },
+    spec: {
+        type: "LoadBalancer",
+        ports: [{ port: 5000, targetPort: "http" }],
+        selector: serverAppLabels,
+        publishNotReadyAddresses: false,
+    }}, {
+    provider: eksCluster.provider,
+    }
+);
+```
+
+And lastly, the client deployment.
+
+```typescript
+const clientAppName = "client";
+const clientAppLabels = { appClass: clientAppName };
+const clientDeployment = new k8s.apps.v1.Deployment("client-side-service", {
+    metadata: { labels: clientAppLabels },
+    spec: {
+        replicas: 2,
+        selector: { matchLabels: clientAppLabels },
+        template: {
+            metadata: { labels: clientAppLabels },
+            spec: {
+                containers: [{
+                    name: clientAppName,
+                    image: awsx.ecr.buildAndPushImage("client-side-service", "./clientside").image(),
+                    ports: [{ name: "http", containerPort: 3000 }],
+                    env: [
+                        { name: "SERVER_HOSTNAME", value: serversideListener.status.loadBalancer.ingress[0].hostname },
+                    ],
+                    resources: {
+                        limits: {
+                            memory: "500Mi",
+                            cpu: "500m"
+                        }
+                    },
+                }],
+            }
+        }
+    }}, {
+    provider: eksCluster.provider,
+});
+
+const clientsideListener = new k8s.core.v1.Service("client-side-listener", {
+    metadata: { labels: clientDeployment.metadata.labels },
+    spec: {
+        type: "LoadBalancer",
+        ports: [{ port: 3000, targetPort: "http" }],
+        selector: clientAppLabels,
+        publishNotReadyAddresses: false,
+    }}, {
+    provider: eksCluster.provider,
+    }
+);
+```
+
+To give us a simple way to connect to our application, we export the clientside listener's address. We can now open a browser window with the URL and port number to view our application.
+
+```typescript
+export const kubeConfig = eksCluster.kubeconfig;
+export const URL = clientsideListener.status.loadBalancer.ingress[0].hostname;
+```
+
+In this example, we demonstrated the benefits that Kubernetes offers, and showed how to use Pulumi to deploy an application to Kubernetes hosted on AWS. Although it might seem uneccessary to use such a system for only a single voting app, the advantages of running your project in the form of containerized services quickly becomes apparent as your application grows in size.
```

Comment:
```suggestion
In this example, we demonstrated the benefits Kubernetes offers and showed how to use Pulumi to deploy an application to Kubernetes hosted on AWS. Although it might seem unnecessary to use such a system for only a single voting app, the advantages of running your project in the form of containerized services become apparent as your application grows in size.
```